src/LlamaFileManager.cpp

```diff
@@ -0,0 +1,648 @@
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+
+#include "LlamaFileManager.h"
+#include <QDebug>
+#include <QStandardPaths>
+#include <QCoreApplication>
+#include <QThread>
+#include <QNetworkProxyFactory>
+#include <QTcpSocket>
+#include <QJsonParseError>
+
+LlamafileManager::LlamafileManager(QObject* parent)
+    : QObject(parent)
+    , process(std::make_unique<QProcess>(this))
+    , healthCheckTimer(std::make_unique<QTimer>(this))
+    , networkManager(std::make_unique<QNetworkAccessManager>(this))
+{
+    // Configure process
+    connect(process.get(), &QProcess::started, this, &LlamafileManager::onProcessStarted);
+    connect(process.get(), QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished),
+            this, &LlamafileManager::onProcessFinished);
+    connect(process.get(), &QProcess::errorOccurred, this, &LlamafileManager::onProcessError);
+    connect(process.get(), &QProcess::stateChanged, this, &LlamafileManager::onProcessStateChanged);
+    
+    // Configure health check timer
+    healthCheckTimer->setSingleShot(false);
+    connect(healthCheckTimer.get(), &QTimer::timeout, this, &LlamafileManager::performHealthCheck);
+    
+    // Configure network manager
+    networkManager->setTransferTimeout(10000); // 10 second timeout
+}
+
+LlamafileManager::~LlamafileManager() {
+    if (isRunning()) {
+        stop();
+    }
+}
+
+bool LlamafileManager::start(const Config& newConfig) {
+    if (currentStatus == Status::Starting || currentStatus == Status::Running) {
+        qDebug() << "LlamafileManager: Already starting or running";
+        return currentStatus == Status::Running;
+    }
+    
+    config = newConfig;
+    
+    if (!validateConfig()) {
+        qDebug() << "LlamafileManager: Invalid configuration: " << lastError;
+        setStatus(Status::Error);
+        return false;
+    }
+    
+    setStatus(Status::Starting);
+    
+    const QString executable = qsl("/bin/sh");
+    const QStringList args = buildProcessArguments();
+    
+    qDebug().noquote() << "Starting llamafile:" << executable << args.join(" ");
+    
+    // Set working directory to the model's directory
+    QFileInfo fileInfo(config.modelPath);
+    if (fileInfo.exists()) {
+        process->setWorkingDirectory(fileInfo.absolutePath());
+    }
+    
+    process->start(executable, args);
+    
+    // Wait for startup with timeout
+    if (!process->waitForStarted(config.startupTimeoutMs)) {
+        const QString error = QString("Failed to start llamafile: %1").arg(process->errorString());
```

Comment:
Hey - I see there are a number of cases in this file where you've used `QString("....")` rather than the agreed/required `qsl("...")` or `tr("...")` wrappers so as to differentiate between texts that do not or DO need to be translated - as such all of these cases will NOT be translated but the absence of the appropriate wrapper means that the intention is not clear.